src/node.cc

```diff
@@ -2234,7 +2234,21 @@ static void Binding(const FunctionCallbackInfo<Value>& args) {
     exports = Object::New(env->isolate());
     DefineConstants(exports);
     cache->Set(module, exports);
-  } else if (!strcmp(*module_v, "natives")) {
+  } else if (!strcmp(*module_v, "$natives") || !strcmp(*module_v, "natives")) {
+    if (!strcmp(*module_v, "natives")) {
+      // graceful-fs < 4 evals process.binding('natives').fs, which prohibits
+      // using internal modules in that module.  Encourage people to upgrade.
+      fprintf(stderr, "(node) process.binding('natives') is deprecated.\n");
+      fprintf(stderr, "(node) If you use graceful-fs < 4, please update.\n");
```

Comment:
While I dislike calling out a specific module here, graceful-fs is really heavily used and is causing specific problems here. Perhaps the compromise is to leave the note in but put a `//TODO: Remove by v6` or similar in there just to mark it as a temporary message?